pkg/service/cachesvc/cache.go

```diff
@@ -79,34 +96,65 @@ func (s *Service) RefreshCachedProviderArgOptions(ctx context.Context, providerI
 		keyers = append(keyers, &op)
 		cachedOpts = append(cachedOpts, op)
 	}
+
+	var cachedGroups []cache.ProviderArgGroupOption
+	if res.Groups != nil {
+		for k, v := range res.Groups.AdditionalProperties {
+			for _, option := range v {
+				op := cache.ProviderArgGroupOption{
+					Provider: providerId,
+					Arg:      argId,
+					Group:    k,
+					Value:    option.Value,
+					Label:    option.Label,
+					Children: option.Children,
+				}
+				keyers = append(keyers, &op)
+				cachedGroups = append(cachedGroups, op)
+			}
+		}
+	}
+
 	err = s.DB.PutBatch(ctx, keyers...)
 	if err != nil {
-		return false, nil, err
+		return false, nil, nil, err
 	}
-	return true, cachedOpts, nil
+	return true, cachedOpts, cachedGroups, nil
 
 }
 
-func (s *Service) fetchProviderOptions(ctx context.Context, providerID, argID string) (types.ArgOptionsResponse, error) {
+func (s *Service) fetchProviderOptions(ctx context.Context, providerID, argID string) (ahtypes.ArgOptionsResponse, error) {
 	res, err := s.AccessHandlerClient.ListProviderArgOptionsWithResponse(ctx, providerID, argID)
 	if err != nil {
-		return types.ArgOptionsResponse{}, err
+		return ahtypes.ArgOptionsResponse{}, err
 	}
 	code := res.StatusCode()
 	switch code {
 	case 200:
-		opts := types.ArgOptionsResponse{
-			HasOptions: res.JSON200.HasOptions,
-			Options:    res.JSON200.Options,
-		}
-		return opts, nil
+		return *res.JSON200, nil
 	case 404:
 		err := errors.New("provider not found")
-		return types.ArgOptionsResponse{}, apio.NewRequestError(err, http.StatusNotFound)
+		return ahtypes.ArgOptionsResponse{}, apio.NewRequestError(err, http.StatusNotFound)
 	case 500:
-		return types.ArgOptionsResponse{}, errors.New(*res.JSON500.Error)
+		return ahtypes.ArgOptionsResponse{}, errors.New(*res.JSON500.Error)
 	default:
 		logger.Get(ctx).Errorw("unhandled access handler response", "response", string(res.Body))
-		return types.ArgOptionsResponse{}, fmt.Errorf("unhandled response code: %d", code)
+		return ahtypes.ArgOptionsResponse{}, fmt.Errorf("unhandled response code: %d", code)
 	}
 }
+
+// Load cached provider argument group option's option.
+func (s *Service) LoadCachedProviderArgGroupOptions(ctx context.Context, providerId string, argId string, groupId string, groupValue string) (bool, cache.ProviderArgGroupOption, error) {
```

Comment:
This only made sense if we wanted to fetch the dynamic values for specific group options only. This could be deleted if we opt to fetch everything related to arg in one single API.